File: issta2018artifacts/minimized-bugs/buggy-clousot/PlumeLib/GetStackTrace.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

// a simplified version of the bug fixed by plume-lib revision 7a6f75bf5a8b72a6ae823d4bd29a43742eb5cf50

class StackTraceElement
{
    public string getClassName() { return null; }
    public string getFileName() { return null; }
    public string getMethodName() { return null; }
    public int getLineNumber() { return 0; }
}

class Throwable
{
    public void fillInStackTrace() { }
    public StackTraceElement[] getStackTrace() { return null; }
}

public class GetStackTrace
{
    public void error()
    {
        Throwable t = new Throwable();
        t.fillInStackTrace();
        StackTraceElement[] ste = t.getStackTrace();
        //:: error: array.access.unsafe.high.constant
        StackTraceElement caller = ste[1];
        Console.Write(
                  "{0}.{1} ({2} line {3})",
                  caller.getClassName(),
                  caller.getMethodName(),
                  caller.getFileName(),
                  caller.getLineNumber());
        for (int ii = 2; ii < ste.Length; ii++)
        {
            Console.Write(" [{0} line {1}]", ste[ii].getFileName(), ste[ii].getLineNumber());
        }
    }
}

File: issta2018artifacts/minimized-bugs/buggy-key/GetStackTrace.java
// a simplified version of the bug fixed by plume-lib revision 7a6f75bf5a8b72a6ae823d4bd29a43742eb5cf50

public class GetStackTrace {
	/*@ public normal_behavior
@ requires true;
@ ensures true;
@*/
    public void error() {
	ThrowableT t = new ThrowableT();
        t.fillInStackTrace();
        StackTraceElementT[] ste = t.getStackTrace();
	//:: error: array.access.unsafe.high.constant
	StackTraceElementT caller = ste[1];
	printf(
			  "%s.%s (%s line %d)",
			  caller.getClassName(),
			  caller.getMethodName(),
			  caller.getFileName(),
			  caller.getLineNumber());
	for (int ii = 2; ii < ste.length; ii++) {
	    printf(" [%s line %d]", ste[ii].getFileName(), ste[ii].getLineNumber());
	}
    }

        /*@ public normal_behavior
@ requires true;
@ ensures true;
@*/
        private static void printf(Object... s) {}

	private class StackTraceElementT {

		/*@ public normal_behavior
@ requires true;
@ ensures true;
@*/
		public Object getClassName() {
			return null;
		}


		/*@ public normal_behavior
@ requires true;
@ ensures true;
@*/
		public Object getMethodName() {
			return null;

		}


		/*@ public normal_behavior
@ requires true;
@ ensures true;
@*/
		public Object getFileName() {
			return null;

		}


		/*@ public normal_behavior
@ requires true;
@ ensures true;
@*/
		public Object getLineNumber() {			return null;

		}
	}

	private class ThrowableT {

		/*@ public normal_behavior
@ requires true;
@ ensures true;
@*/
		public void fillInStackTrace() {

		}

		/*@ public normal_behavior
@ requires true;
@ ensures true;
@*/
		public StackTraceElementT[] getStackTrace() {
			return null;
		}
	}
}
